### DISCUSSION TITLE
Support Ouster lidar

### DISCUSSION THREAD

**Discussion by amacneil** [foxglove/community#746] [Q&A]
via @jtbandes:

We currently support Velodyne packets. We should add support for Ouster too.

See:
- https://github.com/cruise-automation/webviz/issues/162, https://github.com/cruise-automation/webviz/issues/675 (similar webviz feature requests)
- https://data.ouster.io/downloads/software-user-manual/lidar-packet-format-update.pdf
- https://github.com/ouster-lidar/ouster_example

**Comment by amacneil** [comment]
Closing because we have not had any user requests for this. If interested, please comment.

**Comment by amacneil** [comment]
See also: https://github.com/cruise-automation/webviz/issues/714

@albait03 I saw you posted here and then deleted your comment. Did you try your bag file in Foxglove Studio (https://studio.foxglove.dev/)? I'm pretty sure it should work out of the box based on your screenshot.

**Comment by AlexanderBMNdeprecated** [comment]
Thank you for the advice. I tried out Foxglove Studio and it met my requirements. I did not know about Foxglove and that there is an almost identical alternative to Webviz.

**Comment by jtbandes** [comment]
If you'd like to learn more about the differences between Foxglove Studio and Webviz, we have a blog post about it: https://foxglove.dev/blog/foxglove-studio-vs-webviz

**Comment by mamacker** [comment]
We're going to be bringing Ouster Lidar into FoxGlove - just adding a comment to reinforce this need. :)

**Comment by jtbandes** [comment]
Related: foxglove/studio#3076

**Comment by Grisly00** [comment]
Yes please!

**Comment by defunctzombie** [comment]
Is this a separate request from supporting point clouds? We have support for point cloud messages in Studio which you can produce from many different types of lidar sensors. We've no plans to add custom support for ouster lidar beyond point clouds right now.

**Comment by thomasegriffith** [comment]
+1 to the original request.  

I'd argue that for a lot of use cases it's way more efficient from a data footprint and computation perspective to only publish raw packets instead of `PointCloud` or `PointCloud2` messages... and then if one needs to visualize a point cloud, make the visualizer decode the packets into a point cloud.

**Reply by jtbandes** [reply]
Nowadays we generally recommend that folks try a [User Script](https://foxglove.dev/docs/studio/panels/user-scripts) or a [message converter extension](https://foxglove.dev/docs/studio/extensions/guide-create-message-converter) as a way of writing custom code to convert data into a format that Studio supports natively. If someone were interested in working on that I'm sure other folks here would be interested to see the results!

**Reply by thomasegriffith** [reply]
Thanks!  Is [velodyne-cloud](https://github.com/foxglove/velodyne-cloud) a good template for that?

**Reply by jtbandes** [reply]
velodyne-cloud is a standalone package intended to be published on NPM. User scripts and extensions are two different targets that work in different ways – the two docs I linked should get you started but you can also start a new discussion or ask in https://foxglove.dev/slack for assistance if you get stuck with either of those.
